core/src/main/java/hudson/search/FixedSet.java

```diff
@@ -51,7 +51,7 @@ public void find(String token, List<SearchItem> result) {
                 token=token.toLowerCase();
                 name=name.toLowerCase();
             }
-            if(token.equals(i.getSearchName()))
+            if(token.equals(name))
```

Comment:
I see that under the hood `equalsIgnoreCase` uses `regionMatches` is seems to be smarter that just `toLowerCase()` without locale definition. In that case I can remove `toLowerCases` statements and use `equals` and `equalsIgnoreCase` cnditionally.